sklearn/utils/estimator_checks.py

```diff
@@ -293,7 +293,9 @@ def set_testing_parameters(estimator):
     if "decision_function_shape" in params:
         # SVC
         estimator.set_params(decision_function_shape='ovo')
-
+    if "n_best" in params:
```

Comment:
The ``n_best`` parameter (default 3) needs to be less than or equal to ``n_components`` which is 1 in the common tests and hence a ValueError is being raised.